### DISCUSSION TITLE
How the server and client store a persistent state

### DISCUSSION THREAD

**Discussion by wekor** [remix-run/remix#3425] [Q&A]
The loader request API, how to carry a token, how the client request API carries a token, and how the server and client store a persistent state

**Comment by sergiodxa** [comment]
You need to use [cookies](https://remix.run/docs/en/v1/api/remix#cookies) or [session cookies](https://remix.run/docs/en/v1/api/remix#sessions) to keep data, like tokens, between requests. The browser will store the cookie or session cookie and send it on every request to your server automatically.

Ideally, you want your cookie to be configured as httpOnly so client-side JS can't read it, add a secret, enable secure as true in production to require HTTPS, and set sameSite as `lax` which will help mitigate CSRF attacks.
